ayon_server/settings/overrides.py

```diff
@@ -169,42 +169,59 @@ def extract_overrides(
     explicit_pins: list[list[str]] | None = None,
     explicit_unpins: list[list[str]] | None = None,
 ) -> dict[str, Any]:
-    result: dict[str, Any] = {}
+    existing_overrides = existing or {}
+    explicit_pins = explicit_pins or []
+    explicit_unpins = explicit_unpins or []
```

Comment:
I noticed you kept these `lists` instead of `set` likely because the paths you're putting in are `list` and therefore not hashable. However, if we'd make those `tuple` they are hashable and we can get `O1` performance on the lookup?

Reply:
Ah you seem to have commented on this [here](https://github.com/ynput/ayon-backend/pull/253#discussion_r1647372643) already. Ignore me if you don't care. I'm not sure how large these lookup lists tends to be in production use cases so it's hard to say whether it's worth the performance diffs.

Reply:
Usually.... zero. lower 10s tops during settings copy :)